src/utils/definition.js

```diff
@@ -0,0 +1,25 @@
+import Contribution from './contribution'
+import EntitySpec from './entitySpec'
+
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+// Abstract methods for Definition
+export default class Definition {
+  static getPathFromUrl(props) {
+    return props.path ? props.path : props.location ? props.location.pathname.slice(props.match.url.length + 1) : null
+  }
+
+  static getDefinitionPreview(state) {
+    return !state.ui.curate.previewDefinition.isFetching
+      ? Contribution.getChangesFromPreview(
+          Object.assign({}, state.ui.inspect.definition.item),
```

Comment:
do you want just shallow copy here? (not sure where this used to just an honest question)